src/dissectors/ec_o5logon.c

```diff
@@ -62,106 +82,318 @@ FUNC_DECODER(dissector_o5logon)
+               csk = memrchr(csk, 0x40, 20);
+            else
+               csk = memrchr(csk, 0x60, 20);
+
+            if (csk) {
+               csk++;
+               for (i = 0; i < 96; i++) {
+                  while (*csk == ' ')
+                     csk++;
+                  sk[i] = *csk++;
+               }
+               sk[96] = 0;
+               strncpy(conn_status->cli_sk, sk, sizeof(conn_status->cli_sk));
+               conn_status->flags.c_sk = 1;
+#if EC_O5LOGON_DEBUG
+               DISSECT_MSG("O5LOGON: %s:%d->%s:%d Got client session key\n", 
+                     ip_addr_ntoa(&PACKET->L3.src, tmp), 
+                     ntohs(PACKET->L4.src), 
+                     ip_addr_ntoa(&PACKET->L3.dst, tmp), 
+                     ntohs(PACKET->L4.dst));
+#endif
+            }
+#if EC_O5LOGON_DEBUG
+            else {
+               DISSECT_MSG("O5LOGON: %s:%d->%s:%d saw AUTH_SESSKEY but "
+                           "couldn't parse client session key?\n", 
+                     ip_addr_ntoa(&PACKET->L3.src, tmp), 
+                     ntohs(PACKET->L4.src), 
+                     ip_addr_ntoa(&PACKET->L3.dst, tmp), 
+                     ntohs(PACKET->L4.dst));
+            }
+#endif
+         }
       }
-   } else {
-      dissect_create_ident(&ident, PACKET, DISSECT_CODE(dissector_o5logon));
+
+      if (conn_status && conn_status->flags.user && conn_status->flags.s_sk &&
+            conn_status->flags.vfr && conn_status->flags.pw && 
+            conn_status->flags.c_sk) {
+         DISSECT_MSG("O5LOGON: %s@%s:$o5logon$%s*%s*%s*%s\n", 
+               conn_status->user, 
+               ip_addr_ntoa(&conn_status->srv_addr, tmp), 
+               conn_status->srv_sk, 
+               conn_status->salt, 
+               conn_status->pw, 
+               conn_status->cli_sk);
+         dissect_wipe_session(PACKET, DISSECT_CODE(dissector_o5logon));
+      }
+   }
+   else {   /* From server */
 
       if (session_get(&s, ident, DISSECT_IDENT_LEN) == E_SUCCESS) {
-         conn_status = (struct o5logon_status *) s->data;
-         unsigned char *skp  = NULL;
-         unsigned char *saltp = NULL;
-         unsigned char *res = NULL;
-         if (PACKET->DATA.len > 16) {
-            skp  = memmem(ptr, PACKET->DATA.len, "AUTH_SESSKEY", 12);
-            saltp = memmem(ptr, PACKET->DATA.len, "AUTH_VFR_DATA", 13);
-            res = memmem(ptr, PACKET->DATA.len, "invalid username", 16);
+
+         conn_status = (struct o5logon_status*)s->data;
+
+         if (conn_status->flags.s_ano == 0) {
+            if (PACKET->DATA.len > 32) {
+               u_char *ano = memmem(ptr, PACKET->DATA.len, "\xde\xad\xbe\xef", 4);
+               if (ano) {
+                  ano += 25;
+                  conn_status->flags.s_ano = 1;
+#if EC_O5LOGON_DEBUG
+                  DISSECT_MSG("O5LOGON: server ver %u.%u.%u.%u.%u\n", ano[0], 
+                        ano[1] >> 4, ano[1] & 15 , ano[2], ano[3]);
+#endif
```

Comment:
I personally had no need for either of them except while debugging / trying to understand the protocol), but I'm fine with including them w/o debug.